src/main/java/com/github/bot/curiosone/core/analysis/TokenScorer.java

```diff
@@ -21,27 +20,28 @@
    * Calculate the score of the Token.
    *
    * @param t is the Token add t's sentiment's score to scoreSet.
+   * 
+   * @return a double value of the given token, -2.0 if not present.
    */
-  public static Double calculateScore(Token t) {
+  public static double calculateScore(Token t) {
     POS p = t.getMeanings().stream()
         .sorted(comparing(Meaning::getFrequency).reversed())
         .findFirst()
         .get().getPOS();
-
     return (p == POS.ADJ || p == POS.V || p == POS.ADV || p == POS.N)
-        ? dl.getScore(t.getLemma()) : 0.0;
+        ? dl.getScore(t.getLemma()) : DictionaryLoader.INVALID_SCORE;
   }
 
   /**
    * Calculate the score of the input token's list.
    *
    * @param tokenList the token list.
+   * 
+   * @return a double in range -1.0 and 1.0. 
    */
-  public static Double calculateScore(List<Token> tokenList) {
-    Double s = 0.0;
-    for (Token x : tokenList) {
-      s += calculateScore(x);
-    }
-    return s;
+  public static double calculateScore(List<Token> tokenList) {
+    OptionalDouble d = tokenList.stream().mapToDouble(TokenScorer::calculateScore)
```

Comment:
put
`.mapToDouble ` on the next line like `filter`
It will be more readable.

EG:
```
OptionalDouble d = tokenList.stream()
    .mapToDouble(TokenScorer::calculateScore)
    .filter(x -> (x != DictionaryLoader.INVALID_SCORE))
    .average();
```